primap2/_aggregate.py

```diff
@@ -600,3 +693,120 @@ def fill_na_gas_basket_from_contents(
+        -------
+            xr.DataArray with aggregated values for coordinates / dimensions as
+            specified in the agg_info dict
+
+        """
+
+        ds_out = self._ds.copy(deep=True)
+        for var in ds_out.data_vars:
+            ds_out = ds_out.pr.merge(
+                ds_out[var].pr.aggregate_coordinates(
+                    agg_info=agg_info,
+                    tolerance=tolerance,
+                )
+            )
+
+        return ds_out
+
+    def aggregate_entities(
+        self,
+        gas_baskets: dict[str, list[str]],
+    ) -> xr.Dataset:
+        """
+        Creates or fills gas baskets
+        TODO: add logging
+        TODO: add default baskets to primap2 and add a function to aggregate default baskets
+        TODO: rename to something more in line with the other gas basket functions?
+
+        No sanity check on what you aggregate
+
+        Parameters
+        ----------
+
+        gas_baskets
+            dict with the following format
+            gas_baskets = {
+                <basket_name>: <list of basket contents>,
+                ...
+            }
+            example_config = {
+                'FGASES (AR4GWP100)': ['SF6', 'NF3', 'HFCS (AR4GWP100)', 'PFCS (AR4GWP100)'],
+                'KYOTOGHG (AR4GWP100)': ['CO2', 'CH4', 'N2O', 'FGASES (AR4GWP100)'],
+            }
+
+        Returns
+        -------
+        xr.Dataset with aggregated gas baskets
+
+        """
+        ds_out = self._ds.copy(deep=True)
+        entities_present = set(ds_out.data_vars)
+        for basket in gas_baskets.keys():
+            basket_contents_present = [
+                gas for gas in gas_baskets[basket] if gas in entities_present
+            ]
+            if len(basket_contents_present) > 0:
+                if basket in list(ds_out.data_vars):
+                    ds_out[basket] = ds_out.pr.fill_na_gas_basket_from_contents(
+                        basket=basket,
+                        basket_contents=basket_contents_present,
+                        skipna=True,
+                        min_count=1,
+                    )
+                else:
+                    try:
+                        ds_out[basket] = xr.full_like(
+                            ds_out[basket_contents_present[0]], np.nan
+                        ).pr.quantify(units="Gg CO2 / year")
+                        ds_out[basket].attrs = {
+                            "entity": basket.split(" ")[0],
+                            "gwp_context": basket.split(" ")[1][1:-1],
+                        }
+                        ds_out[basket] = ds_out.pr.gas_basket_contents_sum(
+                            basket=basket,
+                            basket_contents=basket_contents_present,
+                            min_count=1,
+                        )
+                        entities_present.add(basket)
+                    except Exception as ex:
+                        print(f"No gas basket created for {basket}: {ex}")
```

Comment:
At a minimum, you have to replace this with `logger.warning` or something, not `print` it.

Reply:
Had planned to do that and forgot.